mlir/lib/Dialect/Tosa/Transforms/TosaFolders.cpp

```diff
@@ -1013,6 +1013,68 @@ struct TosaFoldConstantGreater : public TosaFoldConstantBinary<TosaFoldConstantG
   }
 };
 
+struct TosaFoldConstantBitwiseNot
+    : public TosaFoldConstantUnaryElementwise<TosaFoldConstantBitwiseNot,
+                                              BitwiseNotOp> {
+  using TosaFoldConstantUnaryElementwise<
+      TosaFoldConstantBitwiseNot,
+      BitwiseNotOp>::TosaFoldConstantUnaryElementwise;
+
+  DenseElementsAttr computeInteger(DenseElementsAttr values,
+                                   PatternRewriter &rewriter, TosaOp op) const {
+    return applyElementWise<APInt, APInt, IntegerType>(
+        values,
+        [](const APInt &val, IntegerType) {
+          return APInt(val.getBitWidth(), ~val.getZExtValue());
```

Comment:
We don't need to go APInt -> int -> APInt
```suggestion
          return ~val;
```